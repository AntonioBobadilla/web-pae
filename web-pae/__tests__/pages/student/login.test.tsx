import Login from '@/pages/student/login';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';

describe('Student Login', () => {
  beforeEach(() => {
    render(<Login />);
  });

  it('renders page unchanged', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });

  // it('should display matching error when email is invalid', async () => {
  //   fireEvent.input(screen.getByRole('textbox', { description: 'name' }), {
  //     target: {
  //       value: 'test'
  //     }
  //   });

  //   fireEvent.input(screen.getByRole('textbox', { name: /password/i }), {
  //     target: {
  //       value: 'test'
  //     }
  //   });

  //   fireEvent.submit(screen.getByRole('button'));

  //   expect(screen.getByRole('textbox', { name: /email/i }).value).toBe('test');
  // });
});
