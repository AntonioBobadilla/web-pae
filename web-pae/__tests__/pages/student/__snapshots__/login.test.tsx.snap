// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Login renders tutor login page unchanged 1`] = `
<div>
  <div>
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <main>
      <form
        class="mainContainer"
      >
        <div
          class="leftCont"
        >
          <img
            class="loginImage"
            src="/images/student-login-image.jpg"
          />
        </div>
        <div
          class="signIn"
        >
          <div
            class="paeLogin"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 100px; height: 33px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <img
                alt="logo"
                data-nimg="fixed"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </span>
            <h1
              class="paeText"
            >
               | INICIO
            </h1>
            <h1
              class="title"
            >
              Inicio de sesión
            </h1>
          </div>
          <div
            class="loginFields"
          >
            <div
              class="component"
            >
              <input
                class="textInput"
                id="email"
                name="email"
                placeholder="CORREO INSTITUCIONAL"
                value=""
              />
            </div>
            <div
              class="component"
            >
              <input
                class="textInput"
                id="password"
                name="password"
                placeholder="CONTRASEÑA"
                type="password"
                value=""
              />
            </div>
            <div
              class="componentB"
            >
              <button
                class="container button primary"
                disabled=""
                type="submit"
              >
                INICIAR SESION
              </button>
            </div>
          </div>
          <div
            class="notUser"
          >
            <a
              class="forgotPassword"
              href="/student/forgot-password"
            >
              Olvidé mi contraseña
            </a>
            <h2
              class="register"
            >
              ¿No tienes cuenta?
               
              <a
                class="regLink"
                href="/register"
              >
                Regístrate
              </a>
            </h2>
            <a
              class="privacy"
              href="/"
            >
              Aviso de privacidad
            </a>
          </div>
        </div>
      </form>
    </main>
  </div>
</div>
`;
